Operation decorators

Dawijak04 opened the pull request.

justyna-przy commented on src/main/java/org/jjspizzeria/jjspizzeria/pizza/pizzadecorator/OperationDecorator.java:
You don't need to return null for getDescription. You can either return pizza.getDescription, or make getDescription an abstract method and not to implement a body. I would recommend returning pizza.getDescription because that is what toppingDecorator does and we should be consistent :)

justyna-przy replied:
same with getCost(), just return pizza.getCost() to be consistent. Also small nit, constructors should usually be at the top of the class.

Dawijak04 replied:
I returned 0 and null because thats what you had specified in the diagram. I'll change it now though

justyna-przy commented on src/main/java/org/jjspizzeria/jjspizzeria/pizza/pizzadecorator/SliceDecorator.java:
For each decorator, we need another constructor that doesn't take in Pizza. Reason being is that we need to do checks before we can wrap the pizza in a decorator (see image).
![image](https://github.com/user-attachments/assets/7e4e0fc0-f5d8-4e60-9197-b76e3d26fa66)
This *may* not be needed in the future, since I am thinking of implementing those checks at a GUI level and simply disabling certain buttons when the pizza is in the wrong state. Include it for now because I am honestly getting confused loll and the wine isn't helping.

justyna-przy replied:
oh and also make slice private and maker a getter method please

justyna-przy commented on src/main/java/org/jjspizzeria/jjspizzeria/pizza/pizzadecorator/BakeDecorator.java:
bakeType should be private toooo 👌

justyna-przy reviewed (commented):
Yeah once we get those small changes/improvements should be good for merge, thanks! 🤖

Dawijak04 commented:
I've implemented your requested code changes and also removed getCost() for the concrete decorators, since it's a redundant method if the super class returns pizza.getCost().

justyna-przy reviewed (approved):
Perfect! Sorry about the diagram confusion, I meant that the operationDecorations wouldn't impact cost but represented that as 0. My badd 😅
